scipy/stats/_distn_infrastructure.py

```diff
@@ -11,6 +11,14 @@
 import warnings
 import inspect
 from itertools import zip_longest
+from collections import namedtuple
+
+
+try:
+    import matplotlib.pyplot as plt  # type: ignore[import]
```

Comment:
This is a heavy import, and only used in a single place. Wouldn't it be better to do this inside `def plot`?

Reply:
Good point 👍 We can actually use `spatial._plotutils._held_figure` (we should move this to `_lib`).

Reply:
Sure thing.

Reply:
@tupui could you tell me what `spatial._plotutils._held_figure` does? I'm not familiar with `scipy._lib.decorator`. 

Would I decorate `plot` with `_held_figure` and it would allow the user to pass `ax` but also take care of importing `matplotlib` and creating `ax` if needed?

Reply:
mmmm actually reading this code I am not sure I like this. What I had in mind was closer to statsmodels https://github.com/statsmodels/statsmodels/blob/main/statsmodels/graphics/utils.py

Because the current decorator hides a bit things and should also be re-worked to remove this `hold` thing.
In any case the idea is: have an optional `ax` as a parameter and check in a function if we need to import Matplotlib and create the figure/ax.

But yes, to use it you only need to decorate your function and have `ax` as an optional parameter. But instead of just returning the figure with `ax.figure`.

Another approach is to only work on `ax` (and only returns `ax`) like seaborn seem to be doing. https://github.com/mwaskom/seaborn/blob/069d53a3dd29a965987705c823883a93700bbe5e/seaborn/relational.py#L654

Reply:
OK, please see updates based on what I saw in seaborn.